tests/instruments/hp/test_hp8753e_with_device.py

```diff
@@ -0,0 +1,338 @@
+    assert isinstance(data[12], numpy.complex128)
+    hp8753e.averaging_enabled = True
+    assert hp8753e.averaging_enabled is True
+    hp8753e.averages = 2
+    assert hp8753e.averages == 2
+    hp8753e.scan_single()
+    data = hp8753e.data_complex
+    assert isinstance(data, numpy.ndarray)
+    assert isinstance(data[12], numpy.complex128)
+
+
+def test_hp8753e_with_device_scan(hp8753e):
+    hp8753e.reset()
+    hp8753e.averaging_enabled = False
+    hp8753e.scan(timeout=10)
+    hp8753e.averaging_enabled = True
+    assert hp8753e.averaging_enabled is True
+    hp8753e.averages = 2
+    assert hp8753e.averages == 2
+    hp8753e.scan(timeout=10)
+    data = hp8753e.data_complex
+    assert isinstance(data, numpy.ndarray)
+    assert isinstance(data[12], numpy.complex128)
+
+
+def test_hp8753e_with_device_data_complex(hp8753e):
+    hp8753e.reset()
+    assert hp8753e.averaging_enabled is False
+    hp8753e.trigger_continuous = True
+    assert hp8753e.trigger_continuous is True
+    hp8753e.scan_single()
+
+    data = hp8753e.data_complex
+    assert len(data) == 201
+    assert isinstance(data, numpy.ndarray)
+    assert isinstance(data[12], numpy.complex128)
+
+    # repeat with average of 2 enabled
+    hp8753e.trigger_hold = True
+    assert hp8753e.trigger_hold is True
+    hp8753e.averaging_enabled = True
+    assert hp8753e.averaging_enabled is True
+    hp8753e.averages = 4
+    assert hp8753e.averages == 4
+    hp8753e.averaging_restart()
+    hp8753e.scan(timeout=20)
+
+    data = hp8753e.data_complex
+    assert len(data) == 201
+    assert isinstance(data, numpy.ndarray)
+    assert isinstance(data[12], numpy.complex128)
+
+
+def test_hp8753e_with_device_np_frequencies(hp8753e):
+    hp8753e.reset()
+    numpy.testing.assert_equal(
+        hp8753e.frequencies, numpy.arange(30e3, 6e9 + (6e9 - 30e3) / 200, (6e9 - 30e3) / 200)
+    )
+
+
+def test_hp8753e_with_device_reset(hp8753e):
+    frequency = 500_000_000
+    hp8753e.set_single_frequency_scan(frequency)
+    assert hp8753e.start_frequency == frequency
+    assert hp8753e.stop_frequency == frequency
+    assert hp8753e.scan_points == 3
+    hp8753e.reset()
+    assert hp8753e.start_frequency == 30_000
+    assert hp8753e.stop_frequency == 6_000_000_000
+    assert hp8753e.scan_points == 201
+
+
+def test_hp8753e_with_device_averaging_enabled(hp8753e):
+
+    hp8753e.averaging_enabled = True
+    assert hp8753e.averaging_enabled is True
+    hp8753e.averaging_enabled = False
+    assert hp8753e.averaging_enabled is False
+
+
+def test_hp8753e_with_device_averages(hp8753e):
+
+    hp8753e.averaging_enabled = True
+    hp8753e.averages = 32
+    assert hp8753e.averages == 32
+    hp8753e.averages = 18
+    assert hp8753e.averages == 18
+    hp8753e.averages = 1
+    assert hp8753e.averages == 1
+    hp8753e.averaging_enabled = False
+
+
+# this test will wear out the mechanical relays and should be run only as necessary
+@pytest.mark.skip
```

Comment:
You can add a reason to the skip:
```suggestion
@pytest.mark.skip(reason="This test will wear out the mechanical relays and should be run only as necessary")
```